here-naksha-lib-psql/src/commonTest/kotlin/naksha/psql/TestPsql.kt

```diff
@@ -175,6 +179,56 @@ class TestPsql {
                         }
                 }
         }
+    }
+
+    @Test
+    fun returns_saved_geometry() {
+        // Given: Create collection request
+        val col = NakshaCollection("test_${randomString().lowercase()}")
+        val writeCollectionRequest = WriteRequest()
+        writeCollectionRequest.writes += Write().createCollection(null, col)
+
+        // And: create feature with geo
+        val writeFeaturesReq = WriteRequest()
+        val feature = NakshaFeature().apply {
+            id = "test_feature"
+            geometry = SpPoint(PointCoord(1.0,2.0, 0.0))
+        }
+        writeFeaturesReq.add(
+            Write().createFeature(null, col.id, feature)
+        )
 
+        // When: executing collection write request
+        env.storage.newWriteSession().use { session: IWriteSession ->
+            val response = session.execute(writeCollectionRequest)
+            assertIs<SuccessResponse>(response)
+            session.commit()
+        }
+
+        // And: executing feature write request
+        env.storage.newWriteSession().use { session: IWriteSession ->
+            val response = session.execute(writeFeaturesReq)
+            assertIs<SuccessResponse>(response)
+            session.commit()
+        }
+
+        // Then: feature is retrievable from the collection
+        val retrievedFeature = env.storage.newReadSession().use { session: IReadSession ->
+            val response = session.execute(
+                ReadFeatures().apply {
+                    collectionIds += col.id
+                    featureIds += feature.id
```

Comment:
There was a test instance where I tried this, I remember setting the feature ID here won't affect much, as the docs mentions that features matching these IDs will be additionally added into the response, so this does not work as a filter it seems.

Reply:
But for this test it will work just fine